tests/kernel/fatal/message_capture/src/main.c

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) 2012-2014 Wind River Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <sys/printk.h>
+
+static volatile int expected_reason = -1;
```

Comment:
I don't understand why it has been declared with "volatile" keyword.

Reply:
I copied this line of code from other exception files. It does not seem necessary to add this volatile keyword here. I'll change the code to get rid of it. Thanks very much.

Reply:
the volatile is there because this global is shared between the exception handler and the thread code

Reply:
@andrewboie Thank you for your comment. I'll change it back.